### PR TITLE
Branch deploy

### PR DESCRIPTION
**Pull request by elenagarrone**
# Description

This PR finalises the testing for the branch deployment and adds the latest fixes.

Link to the testing is in a Jira ticket that I will link to in Slack if interested.

### CODE REVIEW

File: .github/workflow-scripts/gh-pages/deploy.js [hunk 1]
```diff
@@ -75,7 +75,17 @@ async function deploy({ branch, sha, repo, owner, docsPath, github, exec }) {
     if (fs.existsSync(docsPath)) {
       // this is to avoid any merge conflict when an environment is redeployed
       console.info(`ℹ️ Cleaning existing ${docsPath}`);
-      await exec.exec('rm', ['-rf', docsPath]);
+      if (branch === 'master') {
+        // for master we want to clean everything from `./docs` with the exception of the directories of the deployed branches
+        const filesToRemove = fs.readdirSync('./docs', { withFileTypes: true }).filter(item => !item.name.startsWith('sb-')).map(({ name }) => name);
+        for (const file of filesToRemove) {
```
Comment (pauleustice) [inline .github/workflow-scripts/gh-pages/deploy.js:81]: We usually put newlines before code blocks

File: .github/workflow-scripts/gh-pages/deploy.js [hunk 2]
```diff
@@ -151,6 +175,34 @@ async function undeploy({ branch, repo, owner, github, exec }) {
         console.log(e);
       }
     }
+
+    const { data: { environments } } = github.rest.repos.getAllEnvironments({
```
Comment (elenagarrone) [inline .github/workflow-scripts/gh-pages/deploy.js:179]: This block of code removes the environments created by the deployments: https://github.com/Legal-and-General/canopy/settings/environments

**Review by Devaney — APPROVED** [review]

File: .github/workflow-scripts/gh-pages/deploy.js (hunk 1 above)
Comment (elenagarrone) [reply .github/workflow-scripts/gh-pages/deploy.js:81]: Updated

**Review by Devaney — APPROVED** [review]

File: .github/workflow-scripts/gh-pages/deploy.js [hunk 3]
```diff
@@ -75,7 +75,18 @@ async function deploy({ branch, sha, repo, owner, docsPath, github, exec }) {
     if (fs.existsSync(docsPath)) {
       // this is to avoid any merge conflict when an environment is redeployed
       console.info(`ℹ️ Cleaning existing ${docsPath}`);
-      await exec.exec('rm', ['-rf', docsPath]);
+      if (branch === 'master') {
+        // for master we want to clean everything from `./docs` with the exception of the directories of the deployed branches
+        const filesToRemove = fs.readdirSync('./docs', { withFileTypes: true }).filter(item => !item.name.startsWith('sb-')).map(({ name }) => name);
```
Comment (pauleustice) [inline .github/workflow-scripts/gh-pages/deploy.js:80]: I don't have a huge problem with it, but it does seem like this file has an identity crisis with using node for reading files and shell scripts for moving them :)

Comment (elenagarrone) [reply .github/workflow-scripts/gh-pages/deploy.js:80]: Please don't make me change it 🤣

**Review by pauleustice — APPROVED** [review]

**Pull request merged**